andela/travel_tool_back
# #160195763  Fix issue with comments not getting fetched

#### What does this PR do?
 Fixes issue with comments not getting fetched along with the details of a request

#### Description of Task to be completed?
- `Eager-load` comments from the getUserRequestDetails controller

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Change into the repo directory - `cd travel_tool_back`
- Check out the branch - `git checkout bg-fix-get-comment-issue-160317720`
- Navigate to [travela](http://travela-staging.andela.com/) in a browser
- Open the developer tools, locate applications tab, copy `jwt-token` in the cookies
- Install dependencies and start the server - `make start`
- Open postman; under `Headers`, create a field called `Authorization` and paste the value of token copied from the browser
- Still in postman, send a POST request to `localhost:5000/api/v1/requests` to create a new request
- Find information about the fields for the above the `POST` request [here](https://github.com/andela/travel_tool_back/pull/14)
- From the response of the created request, copy the value of `requestId`
- Still in postman, send a post request to `localhost:5000/api/v1/comments` with following fields
    - `comment` - the comment you want to post
    - `requestId` - the requestId you copied
- You should have created a new comment successfully.
- To get the list of comments for a request, in postman, make a `GET` request to `localhost:5000/api/v1/requests/requestId`, where requestId is the id of the request you created earlier.

#### Any background context you want to provide?
Connected to this [feature](https://github.com/andela/travel_tool_back/pull/29)

#### What are the relevant pivotal tracker stories?
None

#### Screenshots (if appropriate)
None

#### Questions:
None
